### ISSUE TITLE
[IR Verifier] didn't check if switch case is constant

### ISSUE THREAD

**Issue by DataCorrupted** [llvm/llvm-project#58583]
```ll
define void @foo(i16 %0, i16 %1) {
entry:
  switch i16 %0, label %default [
    i16 %1, label %on_one
    i16 2, label %on_two
  ]

default:                                          ; preds = %entry
  br label %exit

on_one:                                           ; preds = %entry
  br label %exit

on_two:                                           ; preds = %entry
  br label %exit

exit:                                             ; preds = %on_one, %on_two, %default
  ret void
}
```

The program is illegal but can pass `IR Verifier`. (`LLParser' will still report error)

However, if the programmer incorrectly `setOperand` of a switch in memory, `IR Verifier` won't report error.

Will update a patch to align tests in `IR Verifier` and `LLParser` for switch testing.

P.S> Should we add sanitization when `Switch->setOperand(...)` ? Currently, it only tests out of bound, should we also test for type consistency?

**Comment by DataCorrupted** [comment]
See patch https://reviews.llvm.org/D136656

**Comment by DataCorrupted** [comment]
Fixed in [348189880b7bc9513ccdd0ea3a1e255b4d3190e7](https://github.com/llvm/llvm-project/commit/348189880b7bc9513ccdd0ea3a1e255b4d3190e7)
